Tasks/DownloadSecureFile/task.json

```diff
@@ -31,6 +31,14 @@
             "defaultValue": "",
             "required": true,
             "helpMarkDown": "Select the secure file to download to a temporary location on the agent. The file will be cleaned up after the build or release."
+        },
+        {
+            "name": "secureFileDownloadPath",
+            "type": "filePath",
```

Comment:
On further thought, let's make this a "string" input "type". "filePath" input types when empty default to $(System.WorkingDirectory) which needs to be handled differently.

Reply:
isn't that what you want?  What's a better default?

Reply:
it also makes the designer offer a picker.  with a string it's just a box - go figure it out for yourselves

Reply:
We want the default to be the Agent temp directory, not the default working dir.

Reply:
@steve-hawkins: Currently, I am seeing an error "p not supplied" if I leave the secureFileDownloadPath empty. Can you repro that?

If "type" is "filePath", then to check if it was supplied by the user or not, you will have to use filePathSupplied method from the lib  - e.g. https://github.com/Microsoft/vsts-tasks/blob/9162d2fd072a686b4c4f0b8c46c2564e151fbb6f/Tasks/InstallAppleProvisioningProfile/installprovprofile.ts#L18. Since this is a path "filePath" provides a UI picker experience. Please change the default value for "secureFileDownloadPath" to be $(Agent.TempDirectory) and leave the type as "filePath". I think that also makes it transparent to the user.